### PR TITLE
ROX-18572: Add CEF format options to Syslog form

### PR DESCRIPTION
**Pull request by vjwilson**
## Description

For Nakul's addition of an option to fix a backend bug.

This PR is into the branch with the backend changes, to allow for integrated testing

## Checklist
- [ ] Investigated and inspected CI test results

## Testing Performed

Visual testing for now, since this PR is into the branch with the backend changes, to allow for integrated testing

Screen Shot 2023-08-21 at 3 58 33 PM
Screen Shot 2023-08-21 at 3 54 37 PM

### CODE REVIEW

**Comment by connorgorman** [comment]
Can we please add a note about what these two options are in the UI and explain when you may want one over the other?

**Review by pedrottimark — APPROVED** [review]

**Comment by theencee** [comment]
This is the backend change: https://github.com/stackrox/stackrox/pull/7527

**Comment by pedrottimark** [comment]
Although I approved the technical changes to move things forward, I agree with comment by Connor.

**Message format** also seems more descriptive than **Type** as the field label.

**Comment by vjwilson** [comment]
> Can we please add a note about what these two options are in the UI and explain when you may want one over the other?


> Although I approved the technical changes to move things forward, I agree with comment by Connor.
> 
> **Message format** also seems more descriptive than **Type** as the field label.

Ideally, we would hide the legacy option from anyone who wasn't already using it, but I don't want to do that, just in case someone who needs it (has there Syslog workflow set up the old way), doesn't accidentally save the integration with the new setting and require intervention to get back to the old way.

@pedrottimark 

1. I will change the label, but I'd prefer to do it in the parent PR branch, after we test together and see if there are other changes to be made.
2. Likewise, in the parent branch, I can add helper text, like @connorgorman suggests.

I would have just pushed a commit to the parent branch, but I had already started on a different branch, and it would take too much time to coordinate rebasing, so I just PR'ed from my branch to Nakul's.

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/SyslogIntegrationForm.tsx [hunk 1]
```diff
@@ -194,6 +199,36 @@ function SyslogIntegrationForm({
...
+                                            id="CEF-option"
+                                            key='CEF'
+                                            text="CEF"
+                                            isSelected={values.syslog.messageFormat === 'CEF'}
+                                            onChange={() =>
+                                                setFieldValue('messageFormat', 'CEF')
+                                            }
+                                        />
+                                        <ToggleGroupItem
+                                            key='UNDEFINED'
```
Comment (theencee) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/SyslogIntegrationForm.tsx:223]: I changed this to `LEGACY` on @connorgorman 's recommendation.

File: ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/SyslogIntegrationForm.tsx [hunk 2]
```diff
@@ -194,6 +199,36 @@ function SyslogIntegrationForm({
...
+                                            text="CEF (legacy field order)"
```
Comment (theencee) [inline ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/SyslogIntegrationForm.tsx:224]: `CEF (Legacy)` is what I called in changelog. We can change in the parent PR if you wish.

**Review by theencee — APPROVED** [review]

**Pull request merged**